tests/testthat/setup.R

```diff
@@ -35,3 +36,67 @@ set_redactor(
 # Mocks are in directories by Connect version. 2024.08.0 contains all mocks
 # created before 2024.09.0, and is the default mock path.
 .mockPaths("2024.08.0")
+
+MockConnect <- R6Class(
+  "MockConnect",
+  inherit = Connect,
+  public = list(
+    initialize = function(version = NA) {
+      self$server <- "https://connect.example"
+      self$api_key <- "fake"
+      private$.version <- version
+    },
+    request = function(method, url, ..., parser = "parsed") {
+      # Record call
+      self$log_call(paste(method, url))
+
+      # Look for response
+      if (!(url %in% names(self$responses))) {
+        stop("Unexpected URL")
+      }
+      res <- self$responses[[url]]
+
+      if (is.null(parser)) {
+        res
+      } else {
+        self$raise_error(res)
+        httr::content(res, as = parser)
+      }
+    },
+    responses = list(),
+    mock_response = function(path, content, status_code = 200L, headers = c("Content-Type" = "application/json; charset=utf-8")) {
```

Comment:
Obviously you're only doing GETs here so you don't hit this, but it's odd that there's no request method or other stuff here. The assurance that you're making the correct request isn't great, though at least you're checking the request method in the call log assertions.

Reply:
Yeah — I'll want to add a verb argument. The approach of constructing the response list names from `"{verb} {url}"` would work for now — so, looking requests up by the string that's recorded in the `call_log` field.